10371.xml

```diff
@@ -0,0 +1,85 @@
+				<Name>Reboot Error Message</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[Containing the error message of the last reboot failure. This value MUST be set to an empty string on exec on /3/0/4, ideally before to send the response for Reboot resource (3/0/4) to avoid any race conditions]]></Description>
+			</Item>
+			<Item ID="2">
+				<Name>Reset Factory State</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>Integer</Type>
+				<RangeEnumeration>0..3</RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[ 		Containing the state of reset factory operation. At start-up the value should be 0 (initial-state). Once the reset factory process is scheduled (exec on /3/O/5), this value must be set to 1 (resetting), it is advised to do that before to send response of Reset Factory resource (/3/0/5) to avoid any race conditions. If reset fails for any reason value must be set to 2 (reset error). If reset succeeds, state should be set to 3 (reset). If the reset factory need a reboot, reboot state(/10371/0/0) resource SHOULD also be updated accordingly. If reset factory need a reboot, in case of successful reset, the value MUST be set to 3 (reset) at device start-up. If it helps implementation, this is tolerated to have this resource set to 3 (reset) at first device start-up.]]></Description>
+			</Item>
+			<Item ID="3">
+				<Name>Reset Factory Error Message</Name>
+				<Operations>R</Operations>
+				<MultipleInstances>Single</MultipleInstances>
+				<Mandatory>Mandatory</Mandatory>
+				<Type>String</Type>
+				<RangeEnumeration></RangeEnumeration>
+				<Units></Units>
+				<Description><![CDATA[Containing the error message of the last reset factory failure. This value MUST be set to an empty string on exec on /3/0/5, ideally before to send the response for Reset Factory resource (3/0/5) to avoid any race conditions]]></Description>
+			</Item>
+		</Resources>
+		<Description2><![CDATA[This LWM2M object extends Device(3) from LWM2M core specification. It mainly provides features allowing better monitor of Reboot(/3/0/4) and Reset Factory(/3/0/5) operation available on LWM2M Device (3) Object. Without this extension this is really hard to know if a reboot or a reset factory is a success or a failure and why it failed. This is also strongly recommended to use separated response for those resources (/3/0/4 and /3/0/5) and wait for response ACK before to execute/schedule the operation (reboot or reset factory). This prevents LWM2M server to consider the request as a failure in case response is lost. This situation could be annoying as server could decide to resent the request later which could bring to unexpected reboot or reset factory.]]></Description2>
```

Comment:
I use `Description2` for a more complete description. (I don't know if this is the idea)

Reply:
Description 2 is used for a variety of situations. Therefore, it is ok to add further details if needed.

Reply:
> Without this extension this is really hard to know if a reboot or a reset factory is a success [...]

This kind of description sounds weird in an object. Makes sense as part of the discussion here when deciding if/how this should be added, but not as long-term description in an object.

Reply:
The idea was to explain rational of this object.
I thought that it could help people to decide if they need to implement it or not.

Reply:
For that I’d recommend wording like “This object provides additional information on whether xxx succeeded...” etc. “really hard to know” sounds very subjective.